chalicelib/utils.py

```diff
@@ -1,37 +1,49 @@
 #!/usr/bin/env python
 
 import requests
-import pandas as pd
 import io
+import pandas as pd
 
 
 class ManifestIO:
     """
-    Handles a metadata manifest from Boardwalk, created by the
-    user, reformats it, creates a workspace in a namespace of
-    of Broad's FireCloud analysis platform.
+    Handles manifest downloaded from a metadata portal
+    by the user. Has methods that check for workspace existence,
+    creates a new workspace, and uploads two files pointed to in
+    the payload into a namespace on Broad's FireCloud analysis platform.
 
-    :flo: (str) a binary file-like object of the TSV metadata file
+    :payload: (list) of strings representing paths to
+              participant (element [0])
+              and sample (element [1])
     :url: (str) base url for FireCloud (FC) workspaces
     :workspace: (str) FC workspace (does not need to exist)
     :namespace: (str) FC namespace
-    :token: (str) filename of bearer token to authenticate to FC
+    :auth: (str) filename of bearer token to authenticate to FC
+
+    TODO: - check whether list has two elements
+          - check which of those elements contains the participant,
+            and which the sample, such that we do not need to hardcode
+            it
+          - programmatically check identity and validity of files
+            pointed to in payload (independ of naming)
+          - avoid writing to TSVs to disk altogether
     """
 
-    def __init__(self, flo, url, workspace, namespace, token):
-        self.df = pd.read_csv(flo, sep='\t')  # create dataframe
-        self.url = url + '/' + namespace + '/' + workspace
+    def __init__(self, data,
+                 url=None, workspace=None, namespace=None, auth=None):
+        self.data = data
+        if url is not None:
+            self.url = url + '/' + namespace + '/' + workspace
         self.workspace = workspace
         self.namespace = namespace
-        self.token = token  # private
+        self.auth = auth  # private
 
     def workspace_exists(self):
         """Returns true if FireCloud workspace in the specified namespace
         exists, otherwise false.
         """
         headers = dict(Accept='application/json',
-                       Authorization=self.token)
-                       #Authorization='Bearer ' + self.token)
+                       Authorization=self.auth)
         r = requests.get(self.url, headers=headers)
         if r.status_code == 200:
```

Comment:
This may be my typed languages bias, but I would change this to:

```
return r.status_code == 200
```

That way the method will always return a boolean. Now it either returns True or None. But maybe this a Python convention and it will work anyway, so i could be off here.